staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader.go

```diff
@@ -105,6 +105,16 @@ func NewSecure(clientCA string, proxyClientNames []string, nameHeaders []string,
 	return x509request.NewVerifier(opts, headerAuthenticator, sets.NewString(proxyClientNames...)), nil
 }
 
+// CanAuthenticateRequest allows a requestHeaderAuthRequestHandler to satisfy the x509request.VerifiableRequest interface
+func (a *requestHeaderAuthRequestHandler) CanAuthenticateRequest(req *http.Request) bool {
+	name := headerValue(req.Header, a.nameHeaders)
+	if len(name) == 0 {
+		return false
+	}
+	return true
```

Comment:
i'm pretty sure this can work here:
`return len(name) != 0`

Reply:
Yes that's **much** better!  I've pushed a commit to clean it up.